### PR TITLE
Fix index creation

### LINKED ISSUE #21: Error "collection.createIndex" at start

### PR DESCRIPTION
**Pull request by pmcnr-hx**
Fixes #21. See #20.

This is a new branch because @slonofanya's fork repository seems to have been deleted. I've rescued his commits from the pull request branch in our repository so his contribution is noted.

I added an additional commit to use `Array.isArray` instead of `instanceof Array` which more robust and use the project's single statement return style.

### CODE REVIEW

**Comment by slonofanya** [comment]
So, let's merge, yeah?

File: lib/mongoHandler.js [hunk 1]
```diff
@@ -147,16 +147,20 @@ MongoStore._unknownError = function(err) {
...
+  if (!Array.isArray(relationshipAttributeNames) || !relationshipAttributeNames.length) return;
+
   var index = relationshipAttributeNames.reduce(function(partialIndex, name) {
-    partialIndex[name + ".id"] = 1;
+    if (name) {
+      partialIndex.push([name + ".id", 1]);
+    }
+
     return partialIndex;
   }, {});
```
Comment (slonofanya) [inline lib/mongoHandler.js]: `}, {});` --> `}, []);`

**Comment by theninj4** [comment]
:+1: looks good, do it :)

**Comment by pmcnr-hx** [comment]
Merged @slonofanya. Enjoy!

**Pull request merged**